src/systemTest/java/soot/asm/AsmMethodSourceOrigNamesTest.java

```diff
@@ -0,0 +1,88 @@
+package soot.asm;
+
+/*-
+ * #%L
+ * Soot - a J*va Optimization Framework
+ * %%
+ * Copyright (C) 2021 Timothy Hoffman
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ * #L%
+ */
+
+import java.util.ArrayList;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+
+import soot.Body;
+import soot.SootMethod;
+import soot.options.Options;
+import soot.testing.framework.AbstractTestingFramework;
+import soot.validation.CheckInitValidator;
+import soot.validation.ValidationException;
+
+/**
+ * @author Timothy Hoffman
+ */
+@PowerMockIgnore({ "com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*", "org.w3c.*" })
+public class AsmMethodSourceOrigNamesTest extends AbstractTestingFramework {
+
+  @Override
+  protected void setupSoot() {
+    final Options opts = Options.v();
+    opts.set_validate(false);
+    opts.setPhaseOption("jb", "use-original-names:true");
+    opts.setPhaseOption("jb.sils", "enabled:false");
+  }
+
+  @Test
+  public void testWriterToUTF8Buffered1() {
+    final String clazz = "org.apache.xml.serializer.WriterToUTF8Buffered";
+    final String[] params = { "char[]", "int", "int" };
```

Comment:
Can you add the Java source of the classes?

Reply:
These are pre-compiled classes that I got from within the DaCapo benchmark suite. The issue only presents itself because whatever compiler was used to produce these class files generated an inconsistent LocalVariableTable. I did manage to find the source for WriterToUTF8Buffered (from xalan-2.6.0) in an online archive and compiled it with javac 1.8.0_121 and the LocalVariableTable was consistent and so could not be used to really test this fix.

Reply:
How could the class with inconsistent LocalVariableTable be compiled? Can such classes be even verified by JVM?

Reply:
In general: If the code can be verified by the JVM, Soot should support it, even if it's strange / unusual. We have seen all sorts of funny code, especially with obfuscation. We can only safely discard such code if the JVM rejects it.

Reply:
According to the JVM spec, the LocalVariableTable is optional and only used by debuggers to associate local variable names with the local variable indexing used by the bytecode. So I assume no verification of the LVT is performed since it's for debuggers only.

[https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.13](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.13)

Reply:
@tim-hoffman I agree that soot should support if jvm verifies the code. Can you use `javap -l` to the bytecode and paste the local variable table and relevant instructions in code comment with some explanation for a test case to help us understand this inconsistency?